raiden/network/rpc/client.py

```diff
@@ -242,6 +246,14 @@ def __init__(
             # scoped web3 instance is used for all clients
             pass
 
+        client_version = self.web3.version.node
+        if 'Geth' in client_version:
+            self.eth_node = EthClient.GETH
+        elif 'Parity' in client_version:
+            self.eth_node = EthClient.PARITY
+        else:
+            raise RuntimeError('Should never get here. Only parity or geth permitted')
```

Comment:
Can we improve this message and use an specific exception type? The [JSONRPCClient is instantiated](https://github.com/raiden-network/raiden/blob/3c1f8a7a14281b6dea99b58dce267afeb3147f68/raiden/ui/cli.py#L592-L597) quite early on the startup process, because we need it to check if we are synchronized and what the the chain we are using

Reply:
It really should never get here in a real run. We [already](https://github.com/raiden-network/raiden/blob/3c1f8a7a14281b6dea99b58dce267afeb3147f68/raiden/ui/cli.py#L110) check the type of the ethereum node when the cli starts.

Reply:
That is exactly 5 lines after the instantiation that I have linked above.

Reply:
Hahaah off by 5 lines. I thought the check was already being performed so this would be a "Should never happen" error. Will fix this.